File: hakimi-erp/backend/app/api/v1/endpoints/report.py
from fastapi import APIRouter, Depends, Query
from sqlalchemy.orm import Session
from app.core.database import get_db
from app.services.report_service import report_service
from app.schemas.base import ResponseModel
from typing import Optional

router = APIRouter()

@router.get("/sales-performance", response_model=ResponseModel)
def sales_performance(days: int = 30, db: Session = Depends(get_db)):
    return ResponseModel(data=report_service.get_sales_performance(db, days))

@router.get("/financial-summary", response_model=ResponseModel)
def financial_summary(db: Session = Depends(get_db)):
    return ResponseModel(data=report_service.get_financial_summary(db))

@router.get("/financial-detail", response_model=ResponseModel)
def financial_detail(db: Session = Depends(get_db)):
    return ResponseModel(data=report_service.get_financial_detail(db))

@router.get("/delivery-stats", response_model=ResponseModel)
def delivery_stats(db: Session = Depends(get_db)):
    return ResponseModel(data=report_service.get_delivery_stats(db))

@router.get("/dashboard-summary", response_model=ResponseModel)
def dashboard_summary(db: Session = Depends(get_db)):
    return ResponseModel(data=report_service.get_dashboard_summary(db))

File: hakimi-erp/frontend/src/api/modules/report.ts
import { get } from '../request'

const BASE = '/api/v1/reports'

export function fetchSalesPerformance(params?: { days?: number }) {
  return get<any>(`${BASE}/sales-performance`, { params })
}

export function fetchFinancialSummary() {
  return get<any>(`${BASE}/financial-summary`)
}

export function fetchFinancialDetail() {
  return get<any>(`${BASE}/financial-detail`)
}

export function fetchDeliveryStats() {
  return get<any>(`${BASE}/delivery-stats`)
}

export function fetchDashboardSummary() {
  return get<any>(`${BASE}/dashboard-summary`)
}

